File: atcoder/abc310a_order_something_else.hs
import Data.List (sort)
-- vicfred
-- https://atcoder.jp/contests/abc310/tasks/abc310_a
-- implementation
main :: IO ()
main = do
  [n, p, q] <- map read . words <$> getLine :: IO [Int]
  d <- sort . map read . words <$> getLine :: IO [Int]
  print $ min (q + head d) p

File: atcoder/abc310A_order_something_else.cpp
// vicfred
// https://atcoder.jp/contests/abc310/tasks/abc310_a
// implementation
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

int main() {
  int N, P, Q;
  cin >> N >> P >> Q;
  vector<long long> D(N);
  for (int i = 0; i < N; ++i) {
    cin >> D[i];
  }
  long long minima = P;
  for (int i = 0; i < N; ++i) {
    minima = min(minima, Q + D[i]);
  }
  cout << minima << endl;
  return 0;
}
